bats/fb-content-upgrade.bats

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bats
+# vim: ft=sh:sw=2:et
+
+set -o pipefail
+
+load os_helper
+load foreman_helper
+
+setup() {
+  tSetOSVersion
+}
+
+@test "upload package" {
+  wget https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/test_errata_install/emoticons-0.1-2.noarch.rpm -P /tmp
+  hammer -u admin -p changeme repository upload-content --organization="${ORGANIZATION}"\
+    --product="${PRODUCT}" --name="${YUM_REPOSITORY}" --path="/tmp/emoticons-0.1-2.noarch.rpm" | grep -q "Successfully uploaded"
+}
+
+@test "sync repository" {
+  hammer -u admin -p changeme repository synchronize --organization="${ORGANIZATION}" \
+    --product="${PRODUCT}" --name="${YUM_REPOSITORY}"
+}
+
+@test "upload puppet module" {
+  curl -o /tmp/stbenjam-dummy-0.2.0.tar.gz https://forgeapi.puppetlabs.com/v3/files/stbenjam-dummy-0.2.0.tar.gz
+  tFileExists /tmp/stbenjam-dummy-0.2.0.tar.gz && hammer -u admin -p changeme repository upload-content \
+    --organization="${ORGANIZATION}" --product="${PRODUCT}" --name="${PUPPET_REPOSITORY}" \
+    --path="/tmp/stbenjam-dummy-0.2.0.tar.gz" | grep -q "Successfully uploaded"
+}
+
+@test "publish content view" {
+  hammer -u admin -p changeme content-view publish --organization="${ORGANIZATION}" \
+    --name="${CONTENT_VIEW}"
+}
+
+@test "promote content view" {
+  hammer -u admin -p changeme content-view version promote  --organization="${ORGANIZATION}" \
+    --content-view="${CONTENT_VIEW}" --to-lifecycle-environment="${LIFECYCLE_ENVIRONMENT}" --from-lifecycle-environment="Library"
+}
+
+@test "register subscription manager" {
+  if [ -e "/etc/rhsm/ca/candlepin-local.pem" ]; then
+    rpm -e `rpm -qf /etc/rhsm/ca/candlepin-local.pem`
+  fi
+
+  rpm -Uvh http://localhost/pub/katello-ca-consumer-latest.noarch.rpm || true
+  subscription-manager register --force --org="Test_Organization" --activationkey="Test AK" || true
+  subscription-manager status | grep -q "Current"
+}
+
+@test "check content host is registered" {
+  hammer -u admin -p changeme host info --name $(hostname -f)
+}
+
+@test "enable content view repo" {
+  subscription-manager repos --enable="Test_Organization_Test_Product_${YUM_REPOSITORY}" | grep -q "is enabled for this system"
+}
+
+@test "install katello-agent" {
+  tPackageUpgrade katello-agent && tPackageExists katello-agent
+}
+
+@test "start katello-agent" {
+  service goferd status || service goferd start
```

Comment:
I can switch to using that - would prefer to do it in all cases separately though.